convert/convert.go

```diff
@@ -247,7 +246,9 @@ func PairwiseFaToVcf(f []*fasta.Fasta, chr string) []*vcf.Vcf {
 		} else if f[0].Seq[i] != f[1].Seq[i] {
 			pastStart = true
 			if insertion { //catches the case where an insertion, now complete, is followed directly by a snp.
-				answer = append(answer, &vcf.Vcf{Chr: chr, Pos: fasta.AlnPosToRefPos(f[0], insertionAlnPos) + 1, Id: ".", Ref: dna.BaseToString(f[0].Seq[insertionAlnPos]), Alt: []string{dna.BasesToString(f[1].Seq[insertionAlnPos:i])}, Qual: 100.0, Filter: "PASS", Info: "."})
+				if !substitutionsOnly {
+					vcf.WriteVcf(out.File, &vcf.Vcf{Chr: chr, Pos: fasta.AlnPosToRefPos(f[0], insertionAlnPos) + 1, Id: ".", Ref: dna.BaseToString(f[0].Seq[insertionAlnPos]), Alt: []string{dna.BasesToString(f[1].Seq[insertionAlnPos:i])}, Qual: 100.0, Filter: "PASS", Info: ".", Format: []string{"."}})
```

Comment:
This should be out instead of out.File.  By writing directly to the file, you "cut in line" to pass the buffer and the possible gzip compression.  This is also why the header appeared at the end.  It was waiting in the buffer for more to be written, while all the following data was written directly to the file, skipping the buffer.